salt/utils/lazy.py

```diff
@@ -116,3 +118,90 @@ def __iter__(self):
         if not self.loaded:
             self._load_all()
         return iter(self._dict)
+
+
+class _LazyModuleLoader(importlib.util.LazyLoader):  # pylint: disable=W0223
+    _known = {"__file__", "__loader__", "__name__", "__path__", "__spec__"}
+
+    class _LazyModule(importlib.util._LazyModule):
+        def __getattribute__(self, attr):
+            # Avoid triggering load for attributes whose value is already known.
+            # This makes it possible to lazy load a module without eagerly
+            # loading its parent module.
+            if attr in _LazyModuleLoader._known:
+                grandself = super(importlib.util._LazyModule, self)
+                return grandself.__getattribute__(attr)
+            super_getattribute = super().__getattribute__
```

Comment:
I will delete this line next time I rebase onto latest `master` (it was originally added for debugging and currently serves no purpose).

```suggestion
```